cli/templateedit_test.go

```diff
@@ -103,4 +93,33 @@ func TestTemplateEdit(t *testing.T) {
 		assert.Equal(t, template.MaxTTLMillis, updated.MaxTTLMillis)
 		assert.Equal(t, template.MinAutostartIntervalMillis, updated.MinAutostartIntervalMillis)
 	})
+	t.Run("InvalidDisplayName", func(t *testing.T) {
+		t.Parallel()
+		client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+		user := coderdtest.CreateFirstUser(t, client)
+		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+		_ = coderdtest.AwaitTemplateVersionJob(t, client, version.ID)
+		template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID)
+
+		// Test the cli command.
+		cmdArgs := []string{
+			"templates",
+			"edit",
+			template.Name,
+			"--name", template.Name,
+			"--display-name", "a-b-c",
+		}
+		cmd, root := clitest.New(t, cmdArgs...)
+		clitest.SetupConfig(t, client, root)
+
+		err := cmd.Execute()
+
+		require.ErrorContains(t, err, `Validation failed for tag "template_display_name"`)
```

Comment:
Ideally, if possible, test for error type, sentinel error, or a simple `err != nil`. This usually results in easier to maintain code (vs having tests fail due to change in error message).